src/cli/util.ts

```diff
@@ -0,0 +1,164 @@
+  const bindings = await generator.generate(genOptions);
+
+  // Write to disk
+  await writeBindings(outputDir, bindings, overwrite);
+}
+
+/**
+ * Write generated bindings to disk (Node.js only)
+ */
+export async function writeBindings(
+  outputDir: string,
+  bindings: GeneratedBindings,
+  overwrite: boolean,
+): Promise<void> {
+  // Check if output directory exists
+  try {
+    const stat = await fs.stat(outputDir);
+    if (stat.isFile()) {
+      throw new Error(`Output directory path is a file: ${outputDir}`);
+    }
+    if (stat.isDirectory() && !overwrite) {
+      throw new Error(
+        `Output directory already exists and overwrite is false: ${outputDir}`,
+      );
+    }
+    if (overwrite) {
+      await fs.rm(outputDir, { recursive: true, force: true });
+    }
+  } catch (error: any) {
+    if (error.code !== "ENOENT") {
+      throw error;
+    }
+  }
+
+  // Create output directory
+  await fs.mkdir(outputDir, { recursive: true });
+
+  await fs.mkdir(path.join(outputDir, "src"), { recursive: true });
+
+  // Write all files
+  const writePromises = [
+    fs.writeFile(path.join(outputDir, "src/index.ts"), bindings.index),
+    fs.writeFile(path.join(outputDir, "src/client.ts"), bindings.client),
+    fs.writeFile(path.join(outputDir, ".gitignore"), bindings.gitignore),
+    fs.writeFile(path.join(outputDir, "README.md"), bindings.readme),
+    fs.writeFile(path.join(outputDir, "package.json"), bindings.packageJson),
+    fs.writeFile(path.join(outputDir, "tsconfig.json"), bindings.tsConfig),
+  ];
+
+  // Only write types file if it's not empty
+  if (bindings.types && bindings.types.trim() !== "") {
+    writePromises.push(
+      fs.writeFile(path.join(outputDir, "src/types.ts"), bindings.types),
+    );
+  }
+
+  await Promise.all(writePromises);
+}
+
+/**
+ * Fetches contract WASM from local file, network hash, or contract ID
+ */
+export async function fetchWasm(args: WasmFetchArgs): Promise<FetchedContract> {
+  // Validate that exactly one source is provided
+  const sources = [args.wasm, args.wasmHash, args.contractId].filter(Boolean);
+  if (sources.length === 0) {
+    throw new WasmFetchError(
+      "Must provide one of the following: --wasm, --wasm-hash, or --contract-id",
+    );
+  }
+  if (sources.length > 1) {
+    ``;
+    throw new WasmFetchError(
+      "Must provide only one of the following: --wasm, --wasm-hash, or --contract-id",
+    );
+  }
+
+  // Handle local WASM file
+  if (args.wasm) {
+    const buffer = await fromWasmFile(args.wasm);
+    return {
+      contract: { type: "wasm", wasmBytes: buffer },
+      source: { type: "file", path: args.wasm },
+    };
+  }
+
+  // For network sources, validate required network parameters
+  if (!args.rpcUrl) {
+    throw new WasmFetchError(
+      "--rpc-url is required when fetching from network",
+    );
+  }
+  if (!args.networkPassphrase) {
+    throw new WasmFetchError(
+      "--network-passphrase is required when fetching from network",
```

Comment:
Can/should we have sensible defaults for the URL given the passphrase, or prefer a simpler `--network testnet|pubnet|futurenet` option that has these prefilled?